akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -442,6 +551,12 @@ private[remote] class Decoder(
             tickMessageCount = messageCount
             tickTimestamp = now
 
+          case AdvertiseActorRefsCompressionTable ⇒
+            compressions.runNextActorRefAdvertisement() // TODO: optimise these operations, otherwise they stall the hotpath
```

Comment:
So that's the tradeoff we're making. This used to be outside of hotpath, but now that we own the compression in this stage and don't synchronise it has to be done here.